### ISSUE TITLE
V1.1.0 Cannot Build Properly In Android Studio

### ISSUE THREAD

**Issue by qiurenbo** [00-Evan/shattered-pixel-dungeon#886]
Unable to find method 'org.apache.commons.compress.archivers.zip.ZipFile.<init>(Ljava/nio/channels/SeekableByteChannel;)V'
org.apache.commons.compress.archivers.zip.ZipFile.<init>(Ljava/nio/channels/SeekableByteChannel;)V

Gradle's dependency cache may be corrupt (this sometimes occurs after a network connection timeout.)

Re-download dependencies and sync project (requires network)
The state of a Gradle build process (daemon) may be corrupt. Stopping all Gradle daemons may solve this problem.

Stop Gradle build processes (requires restart)
Your project may be using a third-party plugin which is not compatible with the other plugins in the project or the version of Gradle requested by the project.

In the case of corrupt Gradle processes, you can also try closing the IDE and then killing all Java processes.

![image](https://user-images.githubusercontent.com/12886911/147051302-d8445010-5ea1-41df-a7ef-9eb2ce36f631.png)

**Comment by qiurenbo** [comment]
Windows 10

**Comment by fregkos** [comment]
I had the same issue, I tried all the aforementioned, but I gave up.

**Comment by 00-Evan** [comment]
I beleive this wholly unhelpful error message is caused by lacking the proper android build tools versions. Make sure you have those installed.

**Comment by qiurenbo** [comment]
It was hard to finally get the program  running in the Android emulator and on the desktop. I changed two things.

# in build.gradle

\- classpath 'com.android.tools.build:gradle:7.0.3' 
\+ classpath 'com.android.tools.build:gradle:3.5.4'

# in gradle.properties
\-  distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-all.zip 
\+ distributionUrl=https\://services.gradle.org/distributions/gradle-6.2-bin.zip

Enjoy the game!

![a26c1d0af741deaa9a29089811c7d8e](https://user-images.githubusercontent.com/12886911/147202603-57be3562-0ab0-445c-8e1f-f3a855e7078d.png)


![68a8d160c897781a10862ed53e2e0e1](https://user-images.githubusercontent.com/12886911/147202782-064ee93f-dac0-465b-ad80-c8f0b94b4ff2.png)


> I had the same issue, I tried all the aforementioned, but I gave up.

Don't give up brother

**Comment by ofey404** [comment]
@qiurenbo I meet this issue too, and your workaround is awesome!

I am learning libgdx by this project, and thanks for @00-Evan for this wonderful game! I guess I play it for more than a hundred hours, and I'm in Patreon :)

**Comment by 00-Evan** [comment]
Reverting to old gradle versions shouldn't be necessary. I'll see if there's something in project setup that coudl make this more streamlined. Again, I'm pretty sure the reason for this is simply that AS isn't downloading the proper dependancies for some reason.

**Comment by qiurenbo** [comment]
# IntelliJ Setup 
@00-Evan You are quite right. The code works fine with the original configuration of the repository. I didn't understand the essence of the problem. It was actually certain key operations that made the code run. For rookies, there is some extra work to be done.

# Gradle and JDK

Configure gradle7.2 with JDK 11 JVM. Otherwise, an error will be reported.

![image](https://user-images.githubusercontent.com/12886911/147326743-4858d04f-d6d3-4406-969c-403e23c6435a.png)

# Android-related Error

If you get an Android-related error. Create a new file ‘local. properties’ in the project folder, and fill in the SDK location.

![image](https://user-images.githubusercontent.com/12886911/147327732-65c4a9e6-d96e-45f5-afec-f6715ee2aa73.png)

I hope it will help the rookies.

**Comment by fregkos** [comment]
I fetched the latest version of the project, chose to install Amazon's Corretto 11 through Android Studio. Now everything works, even with other JDK version, like Oracle's JDK 8 and 11. I'm not sure how to reproduce the problem I had.

**Comment by 00-Evan** [comment]
Alright let me be more specific about what causes this.

If you don't currently have android build tools installed (the repo currently uses 30.0.2) and then attempt a gradle sync, the following error occurs (this is an except from the full log print out, which is a bit more informative):

```
Starting Gradle Daemon...
Gradle Daemon started in 1 s 387 ms
Checking the license for package Android SDK Build-Tools 30.0.2 in C:\Users\Evan\AppData\Local\Android\Sdk\licenses
License for package Android SDK Build-Tools 30.0.2 accepted.
Preparing "Install Android SDK Build-Tools 30.0.2 (revision: 30.0.2)".

FAILURE: Build failed with an exception.

* What went wrong:
'void org.apache.commons.compress.archivers.zip.ZipFile.<init>(java.nio.channels.SeekableByteChannel)'
```

Essentially, for some reason the JDK version that comes built-in with Android studio is incapable of properly extracting the SDK build tools after downloading them. This may be a side-effect of Google switching to JDK 11 in recent AS versions. Manually using other JDKs to run gradle may work fine.

Installing them manually (via tools -> SDK manager) works properly, and AS's built-in JDK will run just fine after that.

I'll look into how to improve this.

**Comment by qiurenbo** [comment]
Thank you for replying.

**Comment by 00-Evan** [comment]
I'm going to leave this open for now until I'm either able to fix the issue or include a warning about it in the project setup docs

**Comment by 00-Evan** [comment]
I've found the cause of this, seems to be a conflict between the android gradle plugin and roboVM gradle plugins. Moving the robovm plugin declaration into the iOS module fixes this. Will include that in 1.1.2.

**Comment by 00-Evan** [comment]
fixed in commit 1b28cd19b5f59a5a8b7b9acf4c3565b19033f378
